qchem/pennylane_qchem/qchem/hf/dipole.py

```diff
@@ -0,0 +1,312 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for computing the dipole moment.
+"""
+import autograd.numpy as anp
+
+from .basis_data import atomic_numbers
+from .hartree_fock import scf
+from .matrices import moment_matrix
+from .observable import fermionic_observable, qubit_observable
+
+
+def dipole_integrals(mol, core=None, active=None):
+    r"""Return a function that computes the dipole moment integrals over the molecular orbitals.
+
+    These integrals are required to construct the dipole operator in the second-quantized form
+
+    .. math::
+
+        \hat{D} = -\sum_{pq} d_{pq} [\hat{c}_{p\uparrow}^\dagger \hat{c}_{q\uparrow} +
+        \hat{c}_{p\downarrow}^\dagger \hat{c}_{q\downarrow}] -
+        \hat{D}_\mathrm{c} + \hat{D}_\mathrm{n},
+
+    where the coefficients :math:`d_{pq}` are given by the integral of the position operator
+    :math:`\hat{{\bf r}}` over molecular orbitals
+    :math:`\phi`
+
+    .. math::
+
+        d_{pq} = \int \phi_p^*(r) \hat{{\bf r}} \phi_q(r) dr,
+
+    and :math:`\hat{c}^{\dagger}` and :math:`\hat{c}` are the creation and annihilation operators,
+    respectively. The contribution of the core orbitals and nuclei are denoted by
+    :math:`\hat{D}_\mathrm{c}` and :math:`\hat{D}_\mathrm{n}`, respectively.
+
+    The molecular orbitals are represented as a linear combination of atomic orbitals as
+
+    .. math::
+
+        \phi_i(r) = \sum_{\nu}c_{\nu}^i \chi_{\nu}(r).
+
+    Using this equation the dipole moment integral :math:`d_{pq}` can be written as
+
+    .. math::
+
+        d_{pq} = \sum_{\mu \nu} C_{p \mu} d_{\mu \nu} C_{\nu q},
+
+    where :math:`d_{\mu \nu}` is the dipole moment integral over the atomic orbitals and :math:`C`
+    is the molecular orbital expansion coefficient matrix. The contribution of the core molecular
+    orbitals is computed as
+
+    .. math::
+
+        \hat{D}_\mathrm{c} = 2 \sum_{i=1}^{N_\mathrm{core}} d_{ii},
+
+    where :math:`N_\mathrm{core}` is the number of core orbitals.
+
+    Args:
+        mol (Molecule): the molecule object
+        core (list[int]): indices of the core orbitals
+        active (list[int]): indices of the active orbitals
+
+    Returns:
+        function: function that computes the dipole moment integrals in the molecular orbital basis
```

Comment:
Just a thought: 
Should we mention type for function as well? I think it is [Callable](https://www.python.org/dev/peps/pep-0483/#fundamental-building-blocks).

Reply:
Maybe we should. I will consider it for a future PR.